File: 2025-and-before/2233-maximum-product-after-k-increments/2233-maximum-product-after-k-increments.java
class Solution {
    public int maximumProduct(int[] nums, int k) {
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        
        for (int iter : nums) {
            pq.add(iter);
        }
        
        while (k-- > 0) {
            int curr = pq.poll();
            pq.add(curr + 1);
        }
        
        int mod = 1000000007;
        long prod = 1;
        
        while (!pq.isEmpty()) {
            int curr = pq.poll();
            prod = (prod * curr) % mod;
        }
        
        return (int) prod;
    }
}

File: 2025-and-before/2233-maximum-product-after-k-increments/2233-maximum-product-after-k-increments.cpp
class Solution {
public:
    int maximumProduct(vector<int>& nums, int k) {
        
        priority_queue<int,vector<int>,greater<int>>pq(nums.begin(),nums.end());
        
        while(k--)
        {
            int x = pq.top();
            pq.pop();
            x++;
            pq.push(x);
        }
        
        long long int ans=1LL, m= 1e9 +7;
        
        while(pq.size()>0)
        {
            int x = pq.top();
            pq.pop();
            ans= ((ans%m) * (x%m))%m;
        }
        
        return ans;
    }
};